core/src/cluster_info.rs

```diff
@@ -482,40 +482,7 @@ impl ClusterInfo {
             && !ContactInfo::is_valid_address(&contact_info.tpu)
     }
 
-    fn stake_weighted_shuffle<S: std::hash::BuildHasher>(
-        peers: &[ContactInfo],
-        stakes: Option<&HashMap<Pubkey, u64, S>>,
-        rng: ChaChaRng,
-    ) -> Vec<(u64, ContactInfo)> {
-        let (stake_weights, peers_with_stakes): (Vec<_>, Vec<_>) = peers
-            .iter()
-            .map(|c| {
-                let stake = stakes.map_or(0, |stakes| *stakes.get(&c.id).unwrap_or(&0));
-                // For stake weighted shuffle a valid weight is atleast 1. Weight 0 is
-                // assumed to be missing entry. So let's make sure stake weights are atleast 1
-                (cmp::max(1, stake), (stake, c.clone()))
-            })
-            .sorted_by(|(_, (l_stake, l_info)), (_, (r_stake, r_info))| {
-                if r_stake == l_stake {
-                    r_info.id.cmp(&l_info.id)
-                } else {
-                    r_stake.cmp(&l_stake)
-                }
-            })
-            .unzip();
-
-        let shuffle = weighted_shuffle(stake_weights, rng);
-
-        let mut out: Vec<(u64, ContactInfo)> = shuffle
-            .iter()
-            .map(|x| peers_with_stakes[*x].clone())
-            .collect();
-
-        out.dedup();
-        out
-    }
-
-    fn peers_and_stakes<S: std::hash::BuildHasher>(
+    fn sorted_stakes_with_index<S: std::hash::BuildHasher>(
         peers: &[ContactInfo],
         stakes: Option<&HashMap<Pubkey, u64, S>>,
     ) -> Vec<(u64, usize)> {
```

Comment:
Do you think we can move this code
```
        stakes_and_index
            .iter_mut()
            .for_each(|(stake, _)| *stake = cmp::max(1, *stake));
```
inside `map` call? I.e. use default stake 1, and also do `stake = cmp::max(1, stake));`

Reply:
Do you mean line 493? i.e move it into the line where we create the (stake,index) tuple? I think that's doable ? It would like this?
```
                let stake = stakes.map_or(1, |stakes| *stakes.get(&c.id).unwrap_or(&1));

```

Reply:
yes, also do `stake = cmp::max(1, stake));` in next line. That's needed in case stake was present but 0 in the hashmap..

Reply:
got it, i'll push a pr.